source/Features/OverrideDescriptions/BonusDescriptions.cs

```diff
@@ -7,127 +7,145 @@
 using CustomComponents;
 using fastJSON;
 using Localize;
+using Newtonsoft.Json;
 
-namespace MechEngineer.Features.OverrideDescriptions
-{
-    [CustomComponent("BonusDescriptions")]
-    public class BonusDescriptions : SimpleCustomComponent, IAdjustTooltip, IAdjustInventoryElement, IAfterLoad
-    {
-        public string[] Bonuses { get; set; }
+namespace MechEngineer.Features.OverrideDescriptions {
+  public enum BonusDescriptionType { Traits, CriticalEffects };
+  public class BonusDescriptionContent {
+    public string template { get; set; }
+    public string elements { get; set; }
+    public BonusDescriptionContent(string t, string e) { template = t; elements = e; }
+  }
+  public class ExtendedDescription {
+    public List<BonusDescriptionContent> Traits { get; set; }
+    public Dictionary<UnitType, List<BonusDescriptionContent>> CriticalEffects { get; set; }
+    public string Description { get; private set; }
+    public ExtendedDescription(string original) {
+      CriticalEffects = new Dictionary<UnitType, List<BonusDescriptionContent>>();
+      Traits = new List<BonusDescriptionContent>();
+      Description = original;
+    }
+  }
+  public static class DescriptionsHelper {
+    private static Dictionary<string, ExtendedDescription> extendedDescriptions = new Dictionary<string, ExtendedDescription>();
+    public static void AddBonusDescription(this DescriptionDef def, BonusDescriptionType type, UnitType ut, string template, string elements, string originalDescription) {
+      //Control.mod.Logger.LogDebug("AddBonusDescription: "+def.Id+" type:"+type+" unit:"+ut+" template:"+template+" elements:"+elements+" original:"+originalDescription);
+      if (extendedDescriptions.ContainsKey(def.Id) == false) { extendedDescriptions.Add(def.Id, new ExtendedDescription(originalDescription)); };
+      ExtendedDescription descr = extendedDescriptions[def.Id];
+      switch (type) {
+        case BonusDescriptionType.Traits: descr.Traits.Add(new BonusDescriptionContent(template,elements)); break;
+        case BonusDescriptionType.CriticalEffects:
```

Comment:
probably also not needed, better to just have CC fields
comp.Is<BonusDescription>(our var e)
e.CriticalEffect
e.Trait
or e.CriticalEffectsFor(actor)

Reply:
enum wih where to show:
Everywhere
mechlab
combatdesciption
statusdescription

enum actor:
ALL
Mech
Turret
Vehicle